mcnp2cad.cpp

```diff
@@ -153,7 +153,13 @@ class GeometryContext {
   std::string materialName( int mat, double rho ){
     std::string ret;
     std::stringstream formatter;
-    formatter << "mat_" << mat << "_rho_" << rho;
+    if(Gopt.uwuw_names){
+      char rho_formatted [50];
+      sprintf(rho_formatted, "%E", rho);
```

Comment:
Could you not replaced the sprintf with something like

```std::string rho_formated;```
```rho_formatted << std::fixed << std::setprecision(8) << std::scientific << rho```

More verbose, but more std::string-y